### PR TITLE
Refactor ReadTracepoints - make it exception-free

### PR DESCRIPTION
**Pull request by dimitry-**
Test: run ServiceTests under root

### CODE REVIEW

File: src/Service/ServiceUtils.cpp [hunk 1]
```diff
@@ -41,36 +41,57 @@ using ::orbit_elf_utils::ElfFile;
...
+    }
+
+    for (auto it = fs::begin(name_directory_iterator), end = fs::end(name_directory_iterator);
+         it != end; it.increment(error)) {
+      if (error) {
+        return ErrorMessage{absl::StrFormat("Unable to scan \"%s\" directory: %s",
+                                            category_path.string(), error.message())};
+      }
+
+      if (it->path().filename() == "enable" || it->path().filename() == "filter") {
```
Comment (dimitry-) [inline src/Service/ServiceUtils.cpp:86]: looks like this should be check for is_directory instead.

Comment (dimitry-) [reply src/Service/ServiceUtils.cpp]: done.

**Review by danielfenner — APPROVED** [review]
lgtm!

File: src/Service/ServiceUtils.cpp (hunk 1 above)
Comment (danielfenner) [reply src/Service/ServiceUtils.cpp]: I think this is wrong - we need to filter out the "enable" and "filter" files.

Comment (dimitry-) [reply src/Service/ServiceUtils.cpp]: but they are files... we are not counting files, only directories. So if Linux decides to add a file in the category directory - we are still fine.

Comment (danielfenner) [reply src/Service/ServiceUtils.cpp]: Oh, I see. You are right: "enable" and "filter" are currently the only files here - so ingoring files is equivalent to the current behaviour and potentially more stable.

**Review by danielfenner — APPROVED** [review]
lgtm!

**Pull request merged**